### PR TITLE
Did a bit of refactoring using the DRY rule

### PR DESCRIPTION
**Pull request by Sick-Carlito**

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Impressive work so far :clap: :clap: However, some changes needs to be implemented in order to get approval for this milestone. check :arrow_down: 

#### Errors to fix

- When I try to run the game it shows the following error:
```
Traceback (most recent call last):
	2: from bin/main.rb:80:in `<main>'
	1: from bin/main.rb:80:in `new'
bin/main.rb:9:in `initialize': uninitialized constant UserInterface::Methods (NameError)

```

https://github.com/Sick-Carlito/tic-tac-toe/blob/e2f29b6a8285175a022b9d8badc67d54783357d2/bin/main.rb#L8-L11

- The game wont work because you are initializing The object of type Methods without it's arguments.

#### Implementation requirements

- **[REQUIRED]** All classes are kept in `lib` folder. Consequently, you will have to load all the classes in `main.rb`

### That's it :relaxed: When finished, kindly resubmit another code review request form :pray: 

Happy coding :v: 
**Ahmed Mahfoudh**

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Good job :+1: 
Your game still throws the same error. Please check my comment to see how you could fix it.
Screenshot from 2019-11-08 00-14-28
#### [OPTIONAL]
- [ ] Tell the users of an invalid input when the type a number above 9. Currently, your game displays the same message for choosing a taken spot.
Screenshot from 2019-11-08 00-17-04

#### Make those changes and submit for another review.
### Martin Afani

File: bin/main.rb [hunk 1]
```diff
@@ -3,14 +3,14 @@
 # frozen_string_literal: true
 
 require_relative '../lib/methods.rb'
+require_relative '../lib/player.rb'
 
 class UserInterface
   def initialize
-    @game = Methods.new
-    @board = @game.init_board(9)
+    @game = Methods.new(@players)
```
Comment (whiz25) [inline bin/main.rb]: Try changing this line to ```@game = GameManager.new(@players)```

**Comment by Sick-Carlito** [comment]
Oh!! sure. Thanks.

**Comment by Sick-Carlito** [comment]
The message returned summarises everything in one sentence. As you can see, it says the spot has been taken OR you've used a wrong value.

If a user tries to use numbers between 1-9 and the error message is returned, you will automatically know that the spot you are trying to fill has already been taken.

If a user chooses a number that is outside the range of 1-9 i.e. 10, the error message will be raised again and you will know you are using the wrong value. I think that part is clear. Thanks.

**Review by wathigo — APPROVED** [review]
## Approved :smile: 
I have enjoyed playing the game.

**[OPTIONAL]**
* We can add a play again functionality after a win or a draw.

**Let's move on :rocket:** 

_Happy Coding :computer:  
Reviewed by: **Simon Wathigo**_

**Comment by Sick-Carlito** [comment]
Thanks.

**Pull request merged**